src/utils/hooks/beaconHooks.ts

```diff
@@ -1,18 +1,27 @@
 import { NetworkType } from "@airgap/beacon-wallet";
+import { get } from "lodash";
 import { useDispatch } from "react-redux";
 
 import { RawPkh } from "../../types/Address";
 import { useAppSelector } from "../redux/hooks";
-import { DAppConnectionInfo, beaconSlice } from "../redux/slices/beaconSlice";
+import { beaconSlice } from "../redux/slices/beaconSlice";
 
 /**
- * Returns connected account pkh & network by a given dAppId.
- *
- * @param dAppId - generated from dApp public key.
+ * Returns function to get connected accounts stored in {@link beaconSlice} by dAppId.
  */
-export const useGetConnectionInfo = (dAppId: string): DAppConnectionInfo | undefined => {
+export const useGetConnectedAccounts = () => {
   const beaconConnections = useAppSelector(s => s.beacon);
-  return beaconConnections[dAppId];
+
+  return (dAppId: string): RawPkh[] => Object.keys(get(beaconConnections, [dAppId], {}));
+};
+
+/**
+ * Returns function to get connection network type stored in {@link beaconSlice} by dAppId & accountPkh.
+ */
+export const useGetConnectionNetworkType = () => {
+  const beaconConnections = useAppSelector(s => s.beacon);
+
+  return (dAppId: string, accountPkh: RawPkh) => get(beaconConnections, [dAppId, accountPkh], "");
```

Comment:
is it possible that we query a non-existing connection?

Reply:
Not with the current implementation.

We use only dAppId + accountPkh from the beacon slice when we call this function.

Do you think it's better to throw error / return undefined if connection does not exists?

Reply:
I think we can have it like `beaconConnections[dAppId][accountPkh]`. it's clean and does the same stuff